### PR TITLE
Fix for multiple getters

### PR DESCRIPTION
**Pull request by XioNoX**
Fix for issue reported in: https://github.com/napalm-automation/napalm-base/issues/180#issuecomment-277608980
After getting a suggestion of how to fix it from @dbarrosop 

Tested with validation files consisting of either lists (new format) and dicts (old format)

### CODE REVIEW

File: napalm_base/validate.py [hunk 1]
```diff
@@ -131,19 +131,22 @@ def compliance_report(cls, validation_file=None):
...
-        else:
-            key = expected_results.pop("_name", "") or getter
-
-            try:
-                kwargs = expected_results.pop('_kwargs', {})
-                actual_results = getattr(cls, getter)(**kwargs)
-                report[key] = _compare_getter(expected_results, actual_results)
-            except NotImplementedError:
-                report[key] = {"skipped": True, "reason": "NotImplemented"}
+    if isinstance(validation_source, dict):
```
Comment (dbarrosop) [inline napalm_base/validate.py]: Don't do this, we should support only one method.

Comment (XioNoX) [inline napalm_base/validate.py]: Maybe use both methods during a transition period? The issue here is that the necessary "new" method would breaks if people are using files with the "old" one.
I'm happy to remove that part if you think it's not worth it.

Comment (dbarrosop) [inline napalm_base/validate.py]: I think it's ok to break it as supporting both might have interesting and unexpected implications on the recursive methods.

Comment (XioNoX) [inline napalm_base/validate.py]: No problem! patch updated!

**Comment by dbarrosop** [comment]
You will have to fix the CI :)

**Comment by XioNoX** [comment]
CI fixed!
I also added the following tests for _name values:
```yaml
- ping:
    _name: ping1
    _kwargs:
        destination: 8.8.8.8
        source: 192.168.1.1
    success:
        packet_loss: 0
    _mode: strict

- ping:
    _name: ping2
    _kwargs:
        destination: 8.8.8.8
        source: 192.168.1.1
    success:
        packet_loss: 0
    _mode: strict
```
@dbarrosop anything left?

**Comment by dbarrosop** [comment]
Ok, now there are conflicts here... I guess that we can close it anyway as they changes here were also on #203?

**Comment by XioNoX** [comment]
Correct, #203 includes the changes from this PR.

**Comment by dbarrosop** [comment]
Ok, closing this one then. Thanks again for the PR : )

**Pull request closed without merging**